include/fmt/std.h

```diff
@@ -389,6 +391,52 @@ struct formatter<
 #endif
   }
 };
+
+namespace detail {
+
+template <typename T, typename Enable = void>
+struct has_flip : std::false_type {};
+
+template <typename T>
+struct has_flip<T, void_t<decltype(std::declval<T>().flip())>>
+    : std::true_type {};
+
+template <typename T> struct is_bit_reference_like {
+  static constexpr const bool value =
+      std::is_convertible<T, bool>::value &&
+      std::is_nothrow_assignable<T, bool>::value && has_flip<T>::value;
```

Comment:
Can we remove the workaround for libc++ if we drop the `is_nothrow_assignable` check?

Reply:
No. ``__bit_const_reference`` does not contain ``flip`` method.

``is_bit_reference_like`` - is standard compatibility reference.
``__bit_const_reference`` is libc++ non standard issue.